app/global_namespace.go

```diff
@@ -3,34 +3,44 @@ package app
 import (
 	"fmt"
 
-	"github.com/rancher/rancher/pkg/settings"
-
 	"github.com/rancher/rancher/pkg/namespace"
+	"github.com/rancher/rancher/pkg/settings"
 	"github.com/rancher/types/config"
 	"github.com/sirupsen/logrus"
 
 	corev1 "k8s.io/api/core/v1"
 	k8serrors "k8s.io/apimachinery/pkg/api/errors"
 )
 
-func addCattleGlobalNamespace(management *config.ManagementContext) error {
+func addCattleGlobalNamespaces(management *config.ManagementContext) error {
+	if _, err := createNamespace(namespace.GlobalNamespace, management); err != nil {
+		return err
+	}
+
+	if _, err := createNamespace(namespace.NodeTemplateGlobalNamespace, management); err != nil {
+		return err
+	}
+
+	logrus.Debugf("calling sync for driver metadata")
+	management.Management.Settings("").Controller().Enqueue("", settings.RkeMetadataConfig.Name)
+
+	return nil
+}
 
+func createNamespace(namespace string, management *config.ManagementContext) (*corev1.Namespace, error) {
 	lister := management.Core.Namespaces("").Controller().Lister()
 
-	_, err := lister.Get("", namespace.GlobalNamespace)
+	ns, err := lister.Get("", namespace)
 	if k8serrors.IsNotFound(err) {
-		ns := &corev1.Namespace{}
-		ns.Name = namespace.GlobalNamespace
+		ns = &corev1.Namespace{}
+		ns.Name = namespace
 		if _, err := management.Core.Namespaces("").Create(ns); err != nil {
-			return fmt.Errorf("Error creating %v namespace: %v", namespace.GlobalNamespace, err)
+			return nil, fmt.Errorf("error creating %v namespace: %v", namespace, err)
```

Comment:
Check the error to see if it's an alreadyExists error.

Reply:
it already checks if the namespace exists above here